src/libraries/System.Private.CoreLib/src/System/Threading/Wasi/WasiPollWorld.wit.imports.wasi.io.v0_2_0.PollInterop.cs

```diff
@@ -0,0 +1,56 @@
+// Generated by `wit-bindgen` 0.27.0. DO NOT EDIT!
+// <auto-generated />
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace WasiPollWorld.wit.imports.wasi.io.v0_2_0
+{
+    internal static class PollInterop {
+
+        internal static class PollWasmInterop
+        {
+            [DllImport("wasi:io/poll@0.2.0", EntryPoint = "poll"), WasmImportLinkage]
+            internal static extern void wasmImportPoll(nint p0, int p1, nint p2);
+
+        }
+
+        internal  static unsafe uint[] Poll(List<global::WasiPollWorld.wit.imports.wasi.io.v0_2_0.IPoll.Pollable> @in)
+        {
+
+            byte[] buffer = new byte[4 * @in.Count];
```

Comment:
Could this become `stackalloc` ?
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc

Or memory or array/memory pool lease ?
https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.rent?view=net-8.0
https://learn.microsoft.com/en-us/dotnet/api/system.buffers.memorypool-1.rent?view=net-8.0

(not blocker for this PR)

Reply:
https://github.com/bytecodealliance/wit-bindgen/issues/1012